crates/pop-drink/src/errors/mod.rs

```diff
@@ -0,0 +1,45 @@
+//! A set of errors used for testing smart contracts.
+
+use std::fmt::Debug;
+
+pub use drink::{
+	pallet_assets::Error as AssetsError, pallet_balances::Error as BalancesError,
+	pallet_contracts::Error as ContractsError,
+};
+
+pub mod v0;
+
+#[track_caller]
+pub fn assert_runtime_err_inner<VersionedApiError, R, E, RuntimeError>(
+	result: Result<R, E>,
+	expected_error: RuntimeError,
+) where
+	// V: Version of Pop API.
+	VersionedApiError: Into<u32>,
+	// E: Returned Err() value of a method result. Must be convertable to `u32`.
+	E: Into<u32>,
+	// D: Expected RuntimeError.
+	RuntimeError: From<u32> + Into<u32> + Debug,
+{
+	let expected_code: u32 = expected_error.into();
+	let expected_error = RuntimeError::from(expected_code);
```

Comment:
Without this, we need to convert it back it two places. `RuntimeError` can't be `Copy`: 

```rs
if let Err(error) = result {
		let error_code: u32 = error.into();
		if error_code != expected_code {
			panic!(
				r#"assertion `left == right` failed
  left: {:?}
 right: {:?}"#,
				RuntimeError::from(error_code),
			        RuntimeError::from(expected_code)
			);
		}
	} else {
		panic!(
			r#"assertion `left == right` failed
  left: Ok()
 right: {:?}"#,
			RuntimeError::from(expected_code)
		);
	}
```

Reply:
I really don't follow. The `expected_error` is given as `RuntimeError` as a parameter. Why do we need to convert it into a `u32` and back into a `RuntimeError`??

Reply:
- Convert `expected_error -> u32`: We need to convert to u32 to compare between `RuntimeError` and a convertible error `E` type.
- When we call `expected_error.into()`, the `expected_error` is borrowed so we can't use it in the `panic!` because `RuntimeError` does not implement `Copy`.

=> We need to convert it back from the code to get rid of the borrowing issue.

Reply:
What about `let expected_code: u32 = expected_error.clone().into();`

Reply:
Can't be cloned. `expected_error` is `RuntimeError` and `Clone` trait can't be implemented for that type.